### PR TITLE
added option to include CacheIgnoreHeaders for disk_cache module

### PR DESCRIPTION
**Pull request by gitkent**
https://httpd.apache.org/docs/2.2/mod/mod_cache.html#cacheignoreheaders
http://blog.futtta.be/2011/02/23/3-apache-mod_cache-gotchas/

### CODE REVIEW

File: templates/mod/disk_cache.conf.erb [hunk 1]
```diff
@@ -2,3 +2,6 @@ CacheEnable disk /
 CacheRoot "<%= @_cache_root %>"
 CacheDirLevels 2
 CacheDirLength 1
+<% if scope['cache_ignore_headers'] != :undef -%>
+CacheIgnoreHeaders <%= @cache_ignore_headers %>
+<% end -%>
```
Comment (ekohl) [inline templates/mod/disk_cache.conf.erb]: In general all files have a newline at the end.

File: templates/mod/disk_cache.conf.erb [hunk 2]
```diff
@@ -2,3 +2,6 @@ CacheEnable disk /
 CacheRoot "<%= @_cache_root %>"
 CacheDirLevels 2
 CacheDirLength 1
+<% if scope['cache_ignore_headers'] != :undef -%>
```
Comment (ekohl) [inline templates/mod/disk_cache.conf.erb]: I think you can simply use ```<% if @cache_ignore_headers -%>``` here

Comment (gitkent) [reply templates/mod/disk_cache.conf.erb]: I'd looked into this before I use the scope:
https://stackoverflow.com/questions/17096934/check-if-a-variable-is-undef-in-puppet-template

It looks like by simply using if is not sufficient to know that a variable is really undef.

Comment (ekohl) [reply templates/mod/disk_cache.conf.erb]: It's good to note that that answer is from 2013. Puppet has evolved a lot so I'm pretty sure that's no longer needed.

**Comment by david22swan** [comment]
@gitkent please alter the docs to mirror your changes and add test coverage.

**Pull request merged**